### PR TITLE
Don't raise on percing some types of percent strings

### PR DESCRIPTION
**Pull request by KitaitiMakoto**
Hi,

I found `rxgettext` command causes errors when it parses some types of percent strings. It's not preferred behavior although it's not trivial how they should be parsed. So, I just fixed for parser not to raise exceptions.

Could you consider this patch? Feel free to point out non-gettext-style fixture and test method.

Thanks.

### CODE REVIEW

File: test/test_parser.rb [hunk 1]
```diff
@@ -142,6 +142,12 @@ def test_p_
         assert_equal "TRANSLATORS:please translate 'name' in the context of 'program'.\n Hint: the translation should NOT contain the translation of 'program'.", t.extracted_comment
       end
     end
+
+    def test_percent_strings
+      assert_nothing_raised do
```
Comment (kou) [inline test/test_parser.rb]: Normally, we should not use `assert_nothing_raised`. It's a meaningless assertion.

How about adding `test/fixtures/_/percent_strings.rb` with the following content:

```ruby
module Fixtures
  module Method_
    class PercentStrings
      include GetText

      bindtextdomain("_", :path => Helper::Path.locale_path)

      def symbol_array
        _(%i(hello world))
      end

      def symbol
        _(%s(hello world))
      end

      def string
        _(%(hello world))
      end

      def string_array
        _(%w(hello world))
      end

      def execute
        _(%x(echo hello world))
      end
    end
  end
end
```

Only `_(%(hello world))` should be extracted in this case.

And a new test should be added into `test/tools/parser/test_ruby.rb`.

Comment (KitaitiMakoto) [reply test/test_parser.rb]: Thank you for the advice! I updated patches.

File: test/fixtures/_/backtick.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2013-2020  Sutou Kouhei <kou@clear-code.com>
```
Comment (kou) [inline test/fixtures/_/backtick.rb]: You can hold your copyright unless you really want to pass your copyright to me :)

```text
Copyright (C) 2020  Kitaiti Makoto <KitaitiMakoto@gmail.com>
```

File: test/fixtures/_/backtick.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
```
Comment (kou) [inline test/fixtures/_/backtick.rb]: Could you remove this needless magic comment?

File: test/fixtures/_/backtick.rb (hunk 2 above)
Comment (KitaitiMakoto) [reply test/fixtures/_/backtick.rb]: Thank you very much for fair advice. I updated.

File: test/fixtures/_/backtick.rb (hunk 3 above)
Comment (KitaitiMakoto) [reply test/fixtures/_/backtick.rb]: Done.

File: lib/gettext/tools/parser/ruby.rb [hunk 4]
```diff
@@ -162,11 +165,13 @@ def process_on_tstring_content(token, po)
       def process_on_tstring_end(token, po)
         @ignore_next_comma = false
         @string_mark_stack.pop
-        last_string = @string_stack.pop
-        if @current_po_entry and last_string
-          @current_po_entry[@current_po_entry_nth_attribute] =
-            (@current_po_entry[@current_po_entry_nth_attribute] || "") +
-            last_string
+        unless @percent_string_stack.pop
```
[the remark this answers is not in the document]
Comment (KitaitiMakoto) [reply lib/gettext/tools/parser/ruby.rb]: Thank you! I was ashamed of myself on rude programming.

**Comment by KitaitiMakoto** [comment]
@kou Thanks for many advice. I fixed all of what you pointed. Can you review again, please?

**Comment by kou** [comment]
Merged.
Thanks!

**Comment by KitaitiMakoto** [comment]
Thank you!

**Pull request merged**